test/client.jl

```diff
@@ -215,6 +215,28 @@ end
     @test_throws HTTP.IOError HTTP.get("http://httpbin.org/delay/5"; readtimeout=1, retry=false)
 end
 
+@testset "Retry all resolved IP addresses" begin
+    # See issue https://github.com/JuliaWeb/HTTP.jl/issues/672
+    # Bit tricky to test, but can at least be tested if localhost
+    # resolves to both IPv4 and IPv6 by listening to the respective
+    # interface
+    alladdrs = Sockets.getalladdrinfo("localhost")
+    if Sockets.ip"127.0.0.1" in alladdrs && Sockets.ip"::1" in alladdrs
```

Comment:
This is true on GitHub Actions. Is there a better way of forcing IP adresses that is not the first? Is there an easy way to insert a bad IP for some host?